src/components/SingleMessage.jsx

```diff
@@ -0,0 +1,78 @@
+import { formatDistance } from 'date-fns'
+import './SingleMessage.css'
+
+export const SingleMessage = ({ singleMessage, fetchPosts }) => {
+
+    const onLikeIncrease = async (thoughts_id) => {
+
+
+        // Defining options for the fetch API call, specifying that the method should be "POST"
+        const options = {
+            // Specifying the request method as POST
+            method: "POST",
+            // Setting the content type of the request to application/json
+            headers: { "Content-Type": "application/json" },
+        };
+
+        // PROMISE LAND;)
+        // - Making a POST request to the API to like a message, using the message's `_id` property to target the correct message
+        // - Parsing the response from the API as JSON
+        // - Updating the `numLikes` and `liked` state variables and fetching the updated posts
+        // - Logging any errors that occur during the fetch operation to the console
+        // post request:
+        try {
+            const response = await fetch(
+                `https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts/${thoughts_id}/like`,
+                options
+            );
+            // Not processing the data returned from API, as a GET request through fetchPosts() is carried out afterwards, so not needed/unactivated:
+            // const data = await response.json();
+
+            // Calling `fetchPosts` function (passed as prop) to re-fetch posts
+            fetchPosts();
```

Comment:
Instead of refetching all posts consider using a local state to keep track of all the likes each post has received. When the API call is successful you can increase this count by one.